deploy/deploy-gcs.yml

```diff
@@ -185,7 +185,7 @@
         - name: GCS | CSI Driver | Wait for csi-nodeplugin to become available
           command: "{{ kubectl }} -n{{ gcs_namespace }} -ojsonpath={.status.numberAvailable} get daemonset csi-nodeplugin-glusterfsplugin"
           register: result
-          until: result.stdout|int == 3
+          until: result.stdout|int == groups['kube-node']|length
```

Comment:
shouldn't it be  `groups['gcs-node']|length` rather than `groups['kube-node']|length` ?

I think there can be kube nodes which we don't want to make as gcs node. So in this scenario I think it should be gcs-node.

https://github.com/gluster/gcs/blob/master/deploy/examples/inventory-gcs-only.example

@Madhu-1 thoughts?

Reply:
@cloudbehl this should be the `kube-node`, as these are the daemonset which will get deployed on all Kube nodes.
https://github.com/gluster/gcs/blob/master/deploy/examples/inventory-gcs-kubespray.example

Reply:
@Madhu-1 Thanks

but does it means if I have 10 node cluster k8s cluster and I want to use 3 nodes as gcs node then gluster-csi-nodeplugin will be installed on all 10 nodes?

Reply:
@cloudbehl yes, the application pods are going to get deployed on all the 10 days, so we need nodeplugin to run on all nodes.